docs/src/components/CopyToClipboardButton.tsx

```diff
@@ -0,0 +1,74 @@
+import Tooltip, { TooltipProps } from '@mui/material/Tooltip';
+import React, { useState } from 'react';
+import { copyTextToClipboard } from '../shared';
+import { isMobile } from 'react-device-detect';
+import CopyAllIcon from '@mui/icons-material/CopyAll';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+
+type Position = 'bottom' | 'top' | 'left' | 'right';
+type CopyToClipboardProps = Omit<
+    Omit<Omit<Omit<Omit<Omit<TooltipProps, 'placement'>, 'open'>, 'children'>, 'onMouseEnter'>, 'onMouseLeave'>,
+    'title'
+> & {
+    duration?: number;
+    position?: Position;
+    title?: string;
+    copiedPosition?: Position;
+    copiedTitle?: string;
+    copyText: string;
+};
```

Comment:
Hmm... what's the reason for expanding `TooltipProps` here? To follow MUI's tradition, "nested" types are usually defined like this:

```tsx
type CopyToClipboardProps = {
    title?: string;
    duration?: number;
    ... 
    TooltipProps?: TooltipProps;
};
```

And then you pass in any props related to the tooltip in an object, like this:

```tsx
<CopyToClipboard TooltipProps={{ content goes here }) ... />
```